src/ipa_pc/mod.rs

```diff
@@ -275,10 +282,10 @@ impl<G: AffineCurve, D: Digest> InnerProductArgPC<G, D> {
                 if (degree_bound_len + 1) % key_len != 0 { Some(degree_bound_len + 1) } else { None }
             });
 
-            assert_eq!(
-                degree_bound_len.is_some(),
-                commitment.shifted_comm.is_some()
-            );
+            // TODO: check error message
+            if degree_bound_len.is_some() != commitment.shifted_comm.is_some() {
+                Err(Error::Other("Degree bound and shifted commitment are not synced".to_owned()))?
```

Comment:
"Degree bound and shifted commitment must be both either present or not present"